File: NBE_5_7_2_02TEAM/src/main/java/io/twogether/nbe_5_7_2_02team/member/util/mapper/FollowMapper.java
package io.twogether.nbe_5_7_2_02team.member.util.mapper;

import io.twogether.nbe_5_7_2_02team.member.domain.Follow;
import io.twogether.nbe_5_7_2_02team.member.dto.response.FollowCreateResponse;

public class FollowMapper {

    public FollowMapper() {}

    public static FollowCreateResponse toFollowCreateResponse(
            Follow follow, Long followerCount, Long followingCount) {
        return new FollowCreateResponse(
                follow.getFollower().getId(),
                follow.getFollowing().getId(),
                followerCount,
                followingCount);
    }
}

File: NBE_5_7_2_02TEAM/src/main/java/io/twogether/nbe_5_7_2_02team/member/util/mapper/FollowMapper.kt
package io.twogether.nbe_5_7_2_02team.member.util.mapper

import io.twogether.nbe_5_7_2_02team.member.domain.Follow
import io.twogether.nbe_5_7_2_02team.member.dto.request.FollowRequest
import io.twogether.nbe_5_7_2_02team.member.dto.response.FollowCreateResponse

fun Long.toFollowRequest(targetId: Long): FollowRequest = FollowRequest(followerId = this, followingId = targetId)

fun Follow.toFollowCreateResponse(
    followerCount: Long,
    followingCount: Long,
): FollowCreateResponse =
    FollowCreateResponse(
        followerId = this.follower.id,
        followingId = this.following.id,
        followerCount = followerCount,
        followingCount = followingCount,
    )
